### DISCUSSION TITLE
How to delete related items

### DISCUSSION THREAD

**Discussion by shashwattpandeyy** [directus/directus#11143] [Archive]
Hey, I have a collection device that has a many-to-many relationship with directus_users. So this creates a new junction table device_directus_users
So in the collection device, we have a field directus_users. I want to dissociate the user from the device by using a custom API. So by this, it will delete the record from the junction table by using cascading. So earlier we have a $delete flat but now we can do this by UI. 
I'm using this custom endpoint and I want to dissociate the user from the device.

` await itemService.updateOne(device.id, {
      directus_users: [
        {
          id: device.directus_users[0].id,
          directus_users: null 
        }
      ]
    })`  when I'm using this it will only nullify the field in the junction table. I want the same functionality that is in App like the cross[x] one when we click this it is deleting the junction table records. So can we do this is a custom API to dissociate the user??

**Comment by rijkvanzanten** [comment]
When you use the `[x]` in the App, it simply sends an array of the value, excluding the one that was deselected. The API will handle the deselect vs delete operation based on the relationship configuration from that point. You can rely on the same logic by simply passing in empty array in your example:

```js
await itemService.updateOne(device.id, { directus_users: [] })
```
